Extension/src/common.ts

```diff
@@ -557,6 +574,67 @@ export function getInstalledBinaryPlatform(): string | undefined {
     return installedPlatform;
 }
 
+/* Check if the core binaries exists in extension's installation folder */
+export async function checkInstallBinariesExist(): Promise<boolean> {
+    if (!checkInstallLockFile()) {
+        return false;
+    }
+    let installBinariesExist: boolean = true;
+    const info: PlatformInformation = await PlatformInformation.GetPlatformInformation();
+    const packageManager: PackageManager = new PackageManager(info);
+    const packages: Promise<IPackage[]> = packageManager.GetPackages();
+    for (const pkg of await packages) {
+        if (pkg.binaries) {
+            await Promise.all(pkg.binaries.map(async (file: string) => {
+                if (!await checkFileExists(file) &&
+                    !(info.platform === "win32" && await checkFileExists(file + ".exe"))) {
+                    installBinariesExist = false;
+                    console.log(`Extension file ${file} is missing.`);
+                }
+            }));
+        }
+    }
+    return installBinariesExist;
+}
+
+/* Check if the core Json files exists in extension's installation folder */
+export async function checkInstallJasonsExist(): Promise<boolean> {
+    let installJasonsExist: boolean = true;
+    const jsonFiles: string[] = [
+        "bin/msvc.arm32.clang.json",
+        "bin/msvc.arm32.gcc.json",
+        "bin/msvc.arm32.msvc.json",
+        "bin/msvc.arm64.clang.json",
+        "bin/msvc.arm64.gcc.json",
+        "bin/msvc.arm64.msvc.json",
+        "bin/msvc.json",
+        "bin/msvc.x64.clang.json",
+        "bin/msvc.x64.gcc.json",
+        "bin/msvc.x64.msvc.json",
+        "bin/msvc.x86.clang.json",
+        "bin/msvc.x86.gcc.json",
+        "bin/msvc.x86.msvc.json",
+        "debugAdapters/bin/cppdbg.ad7Engine.json"
+    ];
+    await Promise.all(jsonFiles.map(async (file) => {
+        if (!await checkFileExists(path.join(extensionPath, file))) {
+            installJasonsExist = false;
+            console.log(`Extension file ${file} is missing.`);
+        }
+    }));
+    return installJasonsExist;
+}
+
+export function removeInstallLockFile(): boolean {
+    try {
+        fs.unlinkSync(path.join(extensionPath, "install.lock"));
```

Comment:
There is a `unlinkPromise` helper function in common.ts, which is awaitable.  I'd suggest making this function awaitable and calling `await unlinkPromise()` instead of calling `fs.unlinkSync`.